app/src/main/java/org/kiwix/kiwixmobile/di/modules/KiwixModule.kt

```diff
@@ -52,4 +53,9 @@ object KiwixModule {
       sharedPreferenceUtil,
       listOf(MountFileSystemChecker(mountPointProducer), FileWritingFileSystemChecker())
     )
+
+  @Provides
+  @KiwixScope
+  fun providesWiFiP2pManager(context: Context): WifiP2pManager? =
```

Comment:
and a comment here explaining why we use the nullable type

Reply:
[ae6afdf](https://github.com/kiwix/kiwix-android/pull/2498/commits/ae6afdf492c4233521a8a13594e93106b3ca830a)

Reply:
It is on the emulator not exclusively on 27, also there is no assertion happening. "We are forced to use the nullable type because of a crash on our nightly builds running on an emulator API 27 [link to ticket]"

Reply:
Why ain't we considering "nullable return type declaration" as an assertion, just because it has no mention of `assert` or is it something else?

Reply:
yes and nullable isn't an assertion, it is an optional container that enforces null safe access through the type system. Were it to be an assertion it says "this item is null or it isn't" which is useless as an assertion. `!!` is more of an assertion "this is not null or I crash the whole damn application"